src/groups/GroupDetails.tsx

```diff
@@ -0,0 +1,213 @@
+import { SpinnerOverlay, useAutoLoadedData } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import React, { useEffect, useState } from 'react';
+import { DeleteUserModal } from 'src/components/DeleteUserModal';
+import { EditUserModal } from 'src/components/EditUserModal';
+import FooterWrapper from 'src/components/FooterWrapper';
+import {
+  AdminNotifierCheckbox,
+  MemberCard,
+  MemberCardHeaders,
+  NewUserCard,
+  Sort,
+  User,
+} from 'src/components/group-common';
+import { DelayedSearchInput } from 'src/components/input';
+import { NewUserModal } from 'src/components/NewUserModal';
+import { PageBox, PageBoxVariants } from 'src/components/PageBox';
+import { TopBar } from 'src/components/TopBar';
+import { Ajax } from 'src/libs/ajax';
+import { GroupRole } from 'src/libs/ajax/Groups';
+import { reportError } from 'src/libs/error';
+import { getLink } from 'src/libs/nav';
+import { useCancellation } from 'src/libs/react-utils';
+import * as StateHistory from 'src/libs/state-history';
+import * as Style from 'src/libs/style';
+import { textMatch } from 'src/libs/utils';
+
+interface GroupDetailsProps {
+  groupName: string;
+}
+
+interface GroupDetailsData {
```

Comment:
These properties were are updated together in the component, so it made sense to track the state as a single object.